### PR TITLE
feat: use ESM

### PR DESCRIPTION
**Pull request by tripodsan**

### CODE REVIEW

**Review by trieloff — APPROVED** [review]
Boom. I trust that this is a pure refactoring.

File: index.js [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
+
+config();
 
 (async () => {
-  await ConfigUtils.checkNodeVersion();
-  await ConfigUtils.validateDotEnv();
-  new CLI().run(process.argv.slice(2));
+  await checkNodeVersion();
+  await validateDotEnv();
+  await new CLI().run(process.argv.slice(2));
```
Comment (dominique-pfister) [inline index.js:24]: So, this _should_ have been awaited already before?

File: src/cli.js [hunk 2]
```diff
@@ -113,16 +120,7 @@ class CLI {
...
-      console.error('Unknown command: %s\n', cmd);
-      argv.showHelp();
-      process.exit(1);
-    }
     // reset fetch connections so that process can terminate
     fetchContext.reset();
   }
 }
-
-module.exports = CLI;
```
Comment (dominique-pfister) [inline src/cli.js:128]: farewell, `module.exports` 👋

File: index.js (hunk 1 above)
Comment (tripodsan) [reply index.js:24]: yes

**Review by dominique-pfister — APPROVED** [review]

**Comment by tripodsan** [comment]
> Boom. I trust that this is a pure refactoring.

yes and not. I also fixed the problem we had with upgrading to the latest yargs version.

**Comment by dominique-pfister** [comment]
Good to know, btw: a lonely `require` in the middle of code needs to be replaced by `await import` 👍

File: src/cli.js [hunk 3]
```diff
@@ -95,16 +86,32 @@ class CLI {
...
 
-    const ret = logArgs(argv)
+    logArgs(argv)
+      .strictCommands(true)
       .scriptName('hlx')
       .usage('Usage: $0 <command> [options]')
       .parserConfiguration({ 'camel-case-expansion': false })
       .env('HLX')
-      .check(envAwareStrict)
+      .check((a) => envAwareStrict(a, argv.parsed.aliases))
```
Comment (tripodsan) [inline src/cli.js:114]: this fixes the yargs 17.x breaking changes

File: src/cli.js [hunk 4]
```diff
@@ -113,16 +120,7 @@ class CLI {
       .help()
       .parse(args);
 
-    // hack to check if command is valid in non-strict mode
-    const cmd = ret._[0];
-    if (cmd && !(cmd in this._commands)) {
-      console.error('Unknown command: %s\n', cmd);
-      argv.showHelp();
-      process.exit(1);
-    }
```
Comment (tripodsan) [inline src/cli.js:122]: this is not longer needed since yargs 17.x has `strictCommands()`

**Pull request merged**